Gem/Code/Source/Components/NetworkMatchComponent.cpp

```diff
@@ -159,14 +165,68 @@ namespace MultiplayerSample
                 if (highestCoins < aznumeric_cast<int>(state.m_score))
                 {
                     highestCoins = aznumeric_cast<int>(state.m_score);
-                    results.m_winningPlayerName = state.m_playerName;
+
+                    // There is no tie so far.
+                    potentialWinners.clear();
+                    potentialWinners.push_back(state);
+                }
+                else if (highestCoins == aznumeric_cast<int>(state.m_score))
+                {
+                    // A potential tie - decide based on remaining armor later.
+                    potentialWinners.push_back(state);                    
                 }
             }
 
             results.m_playerStates.push_back(state);
         }
 
+        FindWinner(results, potentialWinners);
+
         RPC_EndMatch(results);
+        GetMatchPlayerCoinsComponentController()->ResetAllCoins();
+    }
+
+    void NetworkMatchComponentController::FindWinner(MatchResultsSummary& results,
+        const AZStd::vector<PlayerState>& potentialWinners)
+    {
+        if (potentialWinners.empty())
+        {
+            results.m_winningPlayerName = "No players in the match";
+        }
+        else if (potentialWinners.size() == 1)
+        {
+            results.m_winningPlayerName = potentialWinners.front().m_playerName;
+        }
+        else if (potentialWinners.size() > 1)
+        {
+            // A tie - find the player with the largest armor remaining.
+            AZStd::vector<const PlayerState*> playersTiedByArmor;
+            playersTiedByArmor.push_back(&potentialWinners.front());
+
+            for (const PlayerState& potential : potentialWinners)
+            {
+                if (potential.m_remainingArmor == playersTiedByArmor.front()->m_remainingArmor)
+                {
+                    playersTiedByArmor.push_back(&potential);
+                }
+                else if (potential.m_remainingArmor > playersTiedByArmor.front()->m_remainingArmor)
+                {
+                    playersTiedByArmor.clear();
+                    playersTiedByArmor.push_back(&potential);
+                }
+            }
+
+            if (playersTiedByArmor.size() > 1)
+            {
+                // If multiple players are still tied on armor, randomly choose a player
+                const AZ::u64 randomlyChosenWinnerIndex = GetNetworkRandomComponentController()->GetRandomUint64() % playersTiedByArmor.size();
+                results.m_winningPlayerName = playersTiedByArmor[randomlyChosenWinnerIndex]->m_playerName;
+            }
+            else
+            {
+                results.m_winningPlayerName = playersTiedByArmor.front()->m_playerName;
+            }
```

Comment:
Should there just be a tie state?

Reply:
Darrin's vision is to pick a random winner. But feel free to bring it up.